mllib/src/test/scala/org/apache/spark/ml/feature/StringIndexerSuite.scala

```diff
@@ -299,18 +310,17 @@ class StringIndexerSuite
       .setInputCol("label")
       .setOutputCol("labelIndex")
 
-    val expected = Seq(Set((0, 0.0), (1, 0.0), (2, 2.0), (3, 1.0), (4, 1.0), (5, 0.0)),
-      Set((0, 2.0), (1, 2.0), (2, 0.0), (3, 1.0), (4, 1.0), (5, 2.0)),
-      Set((0, 1.0), (1, 1.0), (2, 0.0), (3, 2.0), (4, 2.0), (5, 1.0)),
-      Set((0, 1.0), (1, 1.0), (2, 2.0), (3, 0.0), (4, 0.0), (5, 1.0)))
+    val expected = Seq(Seq((0, 0.0), (1, 0.0), (2, 2.0), (3, 1.0), (4, 1.0), (5, 0.0)),
```

Comment:
I'd make a little argument that if without shuffling, dataframe transformation will keep row ordering. :)

Reply:
I confirmed this with @cloud-fan 
If use the pattern:
```
Seq(...).toDF().select(...).collect()
```
will use `localRelation` and will always use one partition to do computation. And the output row order will keep the same with the input seq.

and seems many other testcases use similar way.

Reply:
I agree that's correct.  The problem is that people tend to see these patterns and copy them without thinking.  It's best to follow patterns which help other contributors to avoid making mistakes.
I'm OK with leaving it since this issue is scattered throughout MLlib tests.